frontend/src/components/Search/utils.test.ts

```diff
@@ -0,0 +1,177 @@
+import SearchGrammarParser from '@/components/Search/Parser/SearchGrammarParser'
+import { parseSearch, STRING_TYPE } from '@/components/Search/utils'
+
+describe('parseSearch', () => {
+	it('should parse a string correctly', () => {
```

Comment:
There seems to be a weird case where certain characters do not show up in the input: `@$%&` and maybe a few others

Reply:
Addressed that in 6c6eb6a2d7c7564c8e4b90cee486bb94590718ee - thanks for catching!